### DISCUSSION TITLE
v20.0.16

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#2548] [Releases]
## Overview
This is a bugfix release with a couple of really good enhancements.
In the meanwhile we're working on completing the full PHPStan validation, which is allowing us to reformat the whole source code to make it look more beautiful than ever.

Last but not least, we already merged 2 PRs for the upcoming PHP 8.2 support!

## Important things you should check before upgrading
* Added weight for API salesOrderShipmentAddTrack by @luigifab in https://github.com/OpenMage/magento-lts/pull/1377
* Color swatches work with disparate product IDs by @luigifab in https://github.com/OpenMage/magento-lts/pull/2390
* Changed default root dir (composer install) by @sreichel in https://github.com/OpenMage/magento-lts/pull/2401
* Fixed targetNamespace in WS-I Compliant by @leissbua in https://github.com/OpenMage/magento-lts/pull/2405
* Removed unused boxover.js by @fballiano in https://github.com/OpenMage/magento-lts/pull/2435
* Fixed "Is Required" custom option attribute always set "Yes" in PHP8.1 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2441
* Removed unused jQuery 1.10.2, updated jQuery 1.12.1 to 1.12.4 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2442
* Removed hideit input field from Contact form by @ADDISON74 in https://github.com/OpenMage/magento-lts/pull/2398
* Split Mage_All.xml into own module XML files by @tmewes in https://github.com/OpenMage/magento-lts/pull/2455
* Updated jQuery Cycle for rwd theme by @fballiano in https://github.com/OpenMage/magento-lts/pull/2505
* Flow.js updates by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2433
* Enabled website config cache by @fballiano in https://github.com/OpenMage/magento-lts/pull/2355
* Removed more Internet Explorer related libraries by @fballiano in https://github.com/OpenMage/magento-lts/pull/2504
* Fixed broken boxes.css in OM 20.0.15 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2364

## What's Changed
* Use constant for _isAllowed() check by @sreichel in https://github.com/OpenMage/magento-lts/pull/2342
* Revert "Add basic text for Ukraine (#2074)" by @fballiano in https://github.com/OpenMage/magento-lts/pull/2325
* Mage_Catalog_Model_Product_Attribute_Backend_Groupprice_Abstract: avoid loading all websites when using only the current one by @fballiano in https://github.com/OpenMage/magento-lts/pull/2351
* Make Curl Client HTTP/2 compatible by @Schrank in https://github.com/OpenMage/magento-lts/pull/1137
* Block access to all dot files. by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/2349
* Capitalization Adjustment Regarding CamelCase in Method Names by @Sdfendor in https://github.com/OpenMage/magento-lts/pull/2365
* Some microoptimization by @sreichel in https://github.com/OpenMage/magento-lts/pull/2335
* Enclosed error with `<pre>` tag for prettier error print. by @kiatng in https://github.com/OpenMage/magento-lts/pull/2368
* Updated phpstan to 1.8.2 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2367
* Escape product titles in MSRP JavaScript by @discountscott in https://github.com/OpenMage/magento-lts/pull/2366
* Backend grid heads - Align the elements for range and date types by @ADDISON74 in https://github.com/OpenMage/magento-lts/pull/2330
* Force describeTable() to use read DB adapter by @fballiano in https://github.com/OpenMage/magento-lts/pull/2371
* Do not install n98/n98_layouthelper by @sreichel in https://github.com/OpenMage/magento-lts/pull/2373
* Replaced deprecated method getRegionJson() with getRegionJsonByStore(). by @kiatng in https://github.com/OpenMage/magento-lts/pull/2375
* Replaced remaining sizeof alias calls  by @Sdfendor in https://github.com/OpenMage/magento-lts/pull/2369
* Removed DISCLAIMER and change Magento -> OpenMage in header by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2297
* Added label for phpstan cosmetic changes by @sreichel in https://github.com/OpenMage/magento-lts/pull/2384
* Added weight for API salesOrderShipmentAddTrack by @luigifab in https://github.com/OpenMage/magento-lts/pull/1377
* PHPStan/DOCBlock fixes ... by @sreichel in https://github.com/OpenMage/magento-lts/pull/2336
* Updated docs for email addTo() by @sreichel in https://github.com/OpenMage/magento-lts/pull/2382
* Updated phpstan experimental by @sreichel in https://github.com/OpenMage/magento-lts/pull/2386
* Cosmetic changes to Mage_Payment_Model_Method_Abstract::validate() by @fballiano in https://github.com/OpenMage/magento-lts/pull/2388
* Replaced Remaining join Alias Calls  by @Sdfendor in https://github.com/OpenMage/magento-lts/pull/2389
* Hidden empty sub menu by @luigifab in https://github.com/OpenMage/magento-lts/pull/2391
* Removed Thumbs.db file by @luigifab in https://github.com/OpenMage/magento-lts/pull/2394
* Fixed Issue with PHP < 8.1 in composer.json by @ADDISON74 in https://github.com/OpenMage/magento-lts/pull/2378
* Blocked access to various project files by @colinmollenhour in https://github.com/OpenMage/magento-lts/pull/2359
* Color swatches work with disparate product IDs by @luigifab in https://github.com/OpenMage/magento-lts/pull/2390
* Moved Credit Memo button at the end by @luigifab in https://github.com/OpenMage/magento-lts/pull/2392
* Minor fixes on 'filter_condition_callback' method _filterStoreConditi… by @kiatng in https://github.com/OpenMage/magento-lts/pull/2362
* Added ReturnTypeWillChange to various Files catched by code style checker by @Flyingmana in https://github.com/OpenMage/magento-lts/pull/2302
* Phpstan fixes by @sreichel in https://github.com/OpenMage/magento-lts/pull/2396
* Fixed camelCase in lib/Varien by @sreichel in https://github.com/OpenMage/magento-lts/pull/2403
* Changed default root dir (composer install) by @sreichel in https://github.com/OpenMage/magento-lts/pull/2401
* Fixed targetNamespace in WS-I Compliant by @leissbua in https://github.com/OpenMage/magento-lts/pull/2405
* Removed deprecated string interpolation for PHP 8.2 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2408
* Moved files out of root by @sreichel in https://github.com/OpenMage/magento-lts/pull/2414
* Product reviews backend: fixed grid ID typo preventing save button to show by @fballiano in https://github.com/OpenMage/magento-lts/pull/2415
* Added ReturnTypeWillChange to Varien_Data_Collection by @Sdfendor in https://github.com/OpenMage/magento-lts/pull/2409
* Added null check before method_exists by @holozaen in https://github.com/OpenMage/magento-lts/pull/2417
* Removed deprecated utf8_encode() for PHP 8.2 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2404
* Method Annotation Improvement Mage_Catalog_Model_Product_Option_Type_Date  by @Sdfendor in https://github.com/OpenMage/magento-lts/pull/2418
* Correct usage for adding items to ArrayObject by @sreichel in https://github.com/OpenMage/magento-lts/pull/2372
* Removed useless array_column function polyfill by @elidrissidev in https://github.com/OpenMage/magento-lts/pull/2423
* Fixed wishlist share email validation by @sreichel in https://github.com/OpenMage/magento-lts/pull/2425
* Small improvements to openmage backend theme for better readability by @fballiano in https://github.com/OpenMage/magento-lts/pull/2257
* Comments More Typo Fixes Retreive => Retrieve by @Sdfendor in https://github.com/OpenMage/magento-lts/pull/2431
* Removed unused boxover.js by @fballiano in https://github.com/OpenMage/magento-lts/pull/2435
* phpstan: fixed calls to Mage_Core_Block_Abstract by @sreichel in https://github.com/OpenMage/magento-lts/pull/2412
* Added some dev tools & code sniffers by @sreichel in https://github.com/OpenMage/magento-lts/pull/2400
* Updated phstorm meta files by @sreichel in https://github.com/OpenMage/magento-lts/pull/2436
* Added "number" validation to minimum order amount in the backend by @fballiano in https://github.com/OpenMage/magento-lts/pull/2439
* DHL International: PHP8: Cast $dimension before calling round() by @fballiano in https://github.com/OpenMage/magento-lts/pull/2438
* [backport] Added a configuration flag to disable global search in adminhtml by @sreichel in https://github.com/OpenMage/magento-lts/pull/2443
* Fixed "Is Required" custom option attribute always set "Yes" in PHP8.1 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2441
* Fixed wrong setDefaultSort() with two parameters by @sreichel in https://github.com/OpenMage/magento-lts/pull/2446
* Fixed wrong camelCase by @sreichel in https://github.com/OpenMage/magento-lts/pull/2452
* phpstan: fixed call to an undefined method Mage_Core_Helper_Abstract::init() by @sreichel in https://github.com/OpenMage/magento-lts/pull/2453
* Fixed yoda style by @sreichel in https://github.com/OpenMage/magento-lts/pull/2459
* Use short array syntax by @sreichel in https://github.com/OpenMage/magento-lts/pull/2458
* Removed meaningless "enter description here" comments in code by @fballiano in https://github.com/OpenMage/magento-lts/pull/2463
* Removed unused constant GALLERY_IMAGE_TABLE by @fballiano in https://github.com/OpenMage/magento-lts/pull/2460
* Fixed yoda style & short array syntax in app\Mage.php by @sreichel in https://github.com/OpenMage/magento-lts/pull/2465
* Fixed sort direction by @sreichel in https://github.com/OpenMage/magento-lts/pull/2468
* Removed deprecated class Mage_Tag_Block_Customer_Edit by @fballiano in https://github.com/OpenMage/magento-lts/pull/2462
* Use composer-plugin to install phpcs sniffs by @sreichel in https://github.com/OpenMage/magento-lts/pull/2470
* Added loading delay adminhtml by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2426
* Fixed some email template resources by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2476
* Phpstan fixes by @sreichel in https://github.com/OpenMage/magento-lts/pull/2457
* Fixed default sort order for admin catalog search terms, closes #2477 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2480
* Cosmetic changes to backend footer by @fballiano in https://github.com/OpenMage/magento-lts/pull/2482
* Addendum to #2426 - Adminhtml loader by @justinbeaty in https://github.com/OpenMage/magento-lts/pull/2475
* Removed memory_limit checks for better performance by @fballiano in https://github.com/OpenMage/magento-lts/pull/2471
* Fixed typos for "deprecated" word by @fballiano in https://github.com/OpenMage/magento-lts/pull/2487
* phpstan: fixed calls to Mage_Core_Abstract_Helper by @sreichel in https://github.com/OpenMage/magento-lts/pull/2483
* Fixed lowercase for constants true/false/null by @fballiano in https://github.com/OpenMage/magento-lts/pull/2489
* Removed useless returns by @fballiano in https://github.com/OpenMage/magento-lts/pull/2488
* Removed useless variables for return statements by @sreichel in https://github.com/OpenMage/magento-lts/pull/1397
* Removed extra blank lines by @fballiano in https://github.com/OpenMage/magento-lts/pull/2490
* Removed deprecation notice for Mage_Catalog_Model_Resource_Product_Attribute_Backend_Urlkey by @fballiano in https://github.com/OpenMage/magento-lts/pull/2461
* Removed unused jQuery 1.10.2, updated jQuery 1.12.1 to 1.12.4 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2442
* Fixed bug on mini login form. by @kiatng in https://github.com/OpenMage/magento-lts/pull/2377
* Removed hideit input field from Contact form by @ADDISON74 in https://github.com/OpenMage/magento-lts/pull/2398
* Updated README with Redis 7 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2491
* Split Mage_All.xml into own module XML files by @tmewes in https://github.com/OpenMage/magento-lts/pull/2455
* Removed note about old (unexisting) branches from README by @fballiano in https://github.com/OpenMage/magento-lts/pull/2495
* Better wording for adminhtml footer by @fballiano in https://github.com/OpenMage/magento-lts/pull/2486
* phpstan: Fixed calls to Mage_Core_Model_Resource_Helper_Abstract by @sreichel in https://github.com/OpenMage/magento-lts/pull/2494
* Removed unused fastclick.js by @fballiano in https://github.com/OpenMage/magento-lts/pull/2503
* Hotfix sorting be entity ID by @sreichel in https://github.com/OpenMage/magento-lts/pull/2509
* Removed deprecated setLanguageCode javascript function by @fballiano in https://github.com/OpenMage/magento-lts/pull/2472
* Updated phpdocs for Mage::log() by @sreichel in https://github.com/OpenMage/magento-lts/pull/2512
* phpstan: fixed unknown types by @sreichel in https://github.com/OpenMage/magento-lts/pull/2500
* Fixed "should return true but returns false" by @sreichel in https://github.com/OpenMage/magento-lts/pull/2513
* Fixed whitespace in docblock for Mage_Adminhtml by @fballiano in https://github.com/OpenMage/magento-lts/pull/2514
* Removed useless constructors by @fballiano in https://github.com/OpenMage/magento-lts/pull/2515
* Fixed MySql4 classes in phpdoc properties/return by @sreichel in https://github.com/OpenMage/magento-lts/pull/2516
* Error_Processor Refactoring by @Sdfendor in https://github.com/OpenMage/magento-lts/pull/2407
* Removed deprecated code from Mage_Rss by @fballiano in https://github.com/OpenMage/magento-lts/pull/2508
* Deprecate PayPal logo block by @fballiano in https://github.com/OpenMage/magento-lts/pull/2498
* Removed border-top from .block-title class (rwd theme) by @fballiano in https://github.com/OpenMage/magento-lts/pull/2507
* Added docs for multistore error pages by @sreichel in https://github.com/OpenMage/magento-lts/pull/2517
* Removed useless constructors by @fballiano in https://github.com/OpenMage/magento-lts/pull/2520
* Removed resource model wrapper by @sreichel in https://github.com/OpenMage/magento-lts/pull/2510
* Fixed whitespace for docblocks in modules (Mage_A*, Mage_B*) by @fballiano in https://github.com/OpenMage/magento-lts/pull/2522
* Fixed whitespace for docblocks in modules (Mage_C* -> Mage_I*) by @fballiano in https://github.com/OpenMage/magento-lts/pull/2523
* Show product collection limiter only if makes sense by @fballiano in https://github.com/OpenMage/magento-lts/pull/2502
* Fixed whitespace for docblocks in modules (Mage_L* -> Mage_P*) by @fballiano in https://github.com/OpenMage/magento-lts/pull/2525
* Implement the possibility to search for null in backend grids by @digitalpianism in https://github.com/OpenMage/magento-lts/pull/1203
* Fixed whitespace for docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/2528
* Avoid getting lock on frontend for flat catalog indexes by @fballiano in https://github.com/OpenMage/magento-lts/pull/2492
* Fixed whitespace for docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/2530
* Replaced Mysql4 class names by @sreichel in https://github.com/OpenMage/magento-lts/pull/2526
* phpstan: fixes for Varien_Data_Collection by @sreichel in https://github.com/OpenMage/magento-lts/pull/2524
* Updated phpdocs, ref #2524 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2534
* Added info about Mage_PageCache removal to README by @fballiano in https://github.com/OpenMage/magento-lts/pull/2531
* Do not startEnvironmentEmulation for current store by @luigifab in https://github.com/OpenMage/magento-lts/pull/1578
* Updated validation for modelName in Mage_Shipping_Model_Config::_getCarrier() and Mage_Payment_Model_Config::_getMethod() by @eneiasramos in https://github.com/OpenMage/magento-lts/pull/2424
* Fixed whitespace for docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/2535
* Fixed README with information that was only on branch 20.0 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2536
* Fixed admin grid filter by @sreichel in https://github.com/OpenMage/magento-lts/pull/2537
* Bugfix to avoid newline character \n in customFormSubmit URL by @eneiasramos in https://github.com/OpenMage/magento-lts/pull/2518
* phpstan: fixed some unknown types by @sreichel in https://github.com/OpenMage/magento-lts/pull/2538
* Updated phpstan to 1.8.3 by @sreichel in https://github.com/OpenMage/magento-lts/pull/2540
* Updated jQuery Cycle for rwd theme by @fballiano in https://github.com/OpenMage/magento-lts/pull/2505
* Fixed whitespace for docblocks by @fballiano in https://github.com/OpenMage/magento-lts/pull/2542
* Fixed broken grids caused by PR #2545 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2546
* Removed more Internet Explorer related libraries by @fballiano in https://github.com/OpenMage/magento-lts/pull/2504
* Fixed broken boxes.css in OM 20.0.15 by @fballiano in https://github.com/OpenMage/magento-lts/pull/2364

## New Contributors
* @discountscott made their first contribution in https://github.com/OpenMage/magento-lts/pull/2366
* @leissbua made their first contribution in https://github.com/OpenMage/magento-lts/pull/2405

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v20.0.15...v20.0.16

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v20.0.16'>v20.0.16</a>.</em>

**Comment by bryanveIoso** [comment]
Wow! Thank you so much for doing God's work in keeping the real Magento up-to-date!

**Comment by BrianPittVP** [comment]
I think it's important that people are aware of how big a change the API Namespace actually is.  For some, it may be nothing, but for others, it could break your integration.  The release notes for version 20.0.15 suggest that the fix is to change any hardcoded API configurations that use "urn:Magento" and change them to match the current NS.  But that will only work if the 3rd party client that's connecting to your site doesn't care what the namespace is.  Some do!!!  In our case, Listrak was not compatible with OM, and the ONLY reason was that they expected "Magento" for the namespace.  After overloading the core to change the namespace back to "Magento" everything started working again.

My personal feeling is that this should NEVER have been changed in the first place.  There's really no reason for it other than divorcing yourself from the old Magento branding, and changing it seemed to only cause problems and didn't solve anything.  It is what it is, but buyer beware.  If you're upgrading an existing Magento install, pay special attention to any API connections that may break.

**Reply by sreichel** [reply]
> My personal feeling is that this should NEVER have been changed in the first place.

:+1:

**Reply by colinmollenhour** [reply]
Agreed. It seems not too late to change this back to urn:Magento to avoid headaches for anyone else that hasn't been affected yet (or just doesn't know it...).

**Reply by fballiano** [reply]
I'm sorry about the problems this change created, but:
- the namespace was changed in 20.0.15, released 2 august, 5 months ago now
- we already discussed a revert and it was decided not to do that
- **if we revert it now, after 6 months, we'll break everybody's website who previously upgraded and patched**
- I don't want to "divorce from the magento branding" but I don't want to babysit a dieing project until people will replace it, I want OpenMage to have an identity and a future and not be "ah is it magento 1 no? ah ok, who cares then". people are spending hundreds of hours trying to keep it alive, sometimes we may break something but it was discussed and it was made with a good heart.
- if more people were participating maybe we could do a better job but who's checking the PRs? and why didn't you face the problems 5 months ago when it was released? I'm not trying to be polemic, but let's be honest

at this point in time i'm totally against reverting this change.

IMHO openmage will never be "magento" again, check how maybe people are actively contributing daily, there's no way anybody can expect it to be the same as before if nobody is willing to put the work. Sorry, but that's the hard truth. We're all doing the best we can, with the resources we have. We may make mistake, if you don't want to see more of them please participate more and together we'll try to do better.

**Reply by leissbua** [reply]
to change it back and recreate problems would just be dumb. I would close the topic once and for all.

**Reply by leissbua** [reply]
Sad story was to see how ERP companies struggle to get their WSDL updated. It created headaches in SAP and AX companies. Finally we helped them to get it updated in their systems, to be able to close that once and for all.

**Reply by colinmollenhour** [reply]
Thanks for the info @fballiano, I assumed this was a new change and didn't look into it further. I agree with your sentiments, especially since so much time has passed. Keep up the great work!

**Comment by leissbua** [comment]
We fixed several bugs and yet reported why this was bad in first place, but now its done since months.
It is way to late to change that back. It created a hell on earth for us. Breaking industry clients api connections. Dont change that back, as majority has yet found a solution for this, and you will restart that mess if you change back again.

**Comment by BrianPittVP** [comment]
I'm glad to see such great discussion on this topic.  To be clear, I posted this as a heads-up to anyone coming here in the future.  We did run into this problem about a month or so ago when we first launched OM (we were on Magento 1.9 until September of this year).  So I was not a part of the project until recently.  At that time, I actually struggled to find answers to the problem, so it's actually a bit of a surprise to hear that it's a known issue that has caused so much industry headache.

I am not suggesting it be changed back, I agree that would be a PITA.  What's done is done.  However, 5 months isn't actually all that long in development timelines.  Moving from a system you've had for a decade takes time, and customers will be moving from Magento to OM for years to come.  If nothing else, highlighting this concern so that it's easy to find when future customers come here looking for answers is still beneficial :)

**Comment by addison74** [comment]
The change was not made because we wanted to replace the word Magento with Openmage, but it had a solid reason. I asked for explanations at that time and I received them, especially since I didn't like a hardcode solution. I think I was one of those who approved the PR. 

It was known that there would be isssues, that's why a Readme was created on this topic, with the "Changes to SOAP/WSDL" section. When it was reported the solution came quickly and many solved their issue until a new version was released. I'm sorry to hear that in some cases the solution took longer, but it's good for the maintainers to be aware of the changes and if they encounter issues to report them immediately. 

I have made my changes in my stores so far and I actually faced this issue when it was reported. Reverting to this means turning things upside down for everyone who tuned in when 20.0.15 was released.

**Comment by leissbua** [comment]
The bad thing is it was hadcoded and not configurable via system.xml. Reason was that they had issues with code testing, but that from my pov clearly should subordinate. They did not think of there are custom soap calls that use the old URN ending up in mixing URN information in WSDL.

**Reply by addison74** [reply]
I agree with that. The value should be changed at any time and quickly in the Backend to avoid errors when long-term edits are needed in the files.

**Comment by addison74** [comment]
We have at our disposal a powerful information tool for such situations, but it is not used https://github.com/OpenMage/Web_Notifications.

Important changes, new versions, can be announced in the Backend whenever necessary.
